packages/scoped-elements/src/ScopedElementsMixin.js

```diff
@@ -1,194 +1,105 @@
- *
- * @param {TemplateResult} template
- * @param {ScopedElementsMap} scopedElements
- * @param {Cache<TemplateStringsArray, TemplateStringsArray>} templateCache
- * @param {Cache<string, string>} tagsCache
- * @returns {TemplateResult}
- */
-const transformTemplate = (template, scopedElements, templateCache, tagsCache) =>
-  new TemplateResult(
-    transform(template.strings, scopedElements, templateCache, tagsCache),
-    transformArray(template.values, scopedElements, templateCache, tagsCache),
-    template.type,
-    template.processor,
-  );
-
-/**
- * Gets an instance of the ScopedElementsTemplateFactory
- *
- * @param {string} scopeName
- * @param {ScopedElementsMap} scopedElements
- * @param {Cache<TemplateStringsArray, TemplateStringsArray>} templateCache
- * @param {Cache<string, string>} tagsCache
- * @returns {function(any): any}
- */
-const scopedElementsTemplateFactory = (
-  scopeName,
-  scopedElements,
-  templateCache,
-  tagsCache,
-) => template => {
-  const newTemplate = transformTemplate(template, scopedElements, templateCache, tagsCache);
-
-  return shadyTemplateFactory(scopeName)(newTemplate);
-};
-
-/** @type {ScopedElementsMixin} */
 const ScopedElementsMixinImplementation = superclass =>
+  /** @type {ScopedElementsMixin} */
   class ScopedElementsHost extends superclass {
     /**
-     * Obtains the scoped elements definitions map
+     * Obtains the scoped elements definitions map if specified.
      *
      * @returns {ScopedElementsMap}
      */
     static get scopedElements() {
       return {};
     }
 
-    /** @override */
-    static render(template, container, options) {
-      if (!options || typeof options !== 'object' || !options.scopeName) {
-        throw new Error('The `scopeName` option is required.');
-      }
-      const { scopeName, eventContext } = options;
-
-      const templateCache = getTemplateCache(eventContext);
-      const tagsCache = getTagsCache(eventContext);
-      const { scopedElements } = this;
-
-      return super.render(template, container, {
-        ...options,
-        templateFactory: scopedElementsTemplateFactory(
-          scopeName,
-          scopedElements,
-          templateCache,
-          tagsCache,
-        ),
-      });
+    /** @type {ShadowRootInit} */
+    static get shadowRootOptions() {
+      return this.__shadowRootOptions;
     }
 
-    /**
-     * Defines a scoped element
-     *
-     * @param {string} tagName
-     * @param {typeof HTMLElement} klass
-     */
-    defineScopedElement(tagName, klass) {
-      return defineScopedElement(tagName, klass, getTagsCache(this));
+    /** @type {ShadowRootInit} */
+    static set shadowRootOptions(value) {
+      this.__shadowRootOptions = value;
+    }
+
+    /** @type {CSSResultFlatArray} */
+    static get elementStyles() {
+      return this.__elementStyles;
+    }
+
+    static set elementStyles(styles) {
+      this.__elementStyles = styles;
```

Comment:
why do we need these getters/setters in this mixin?

Reply:
for typescript. would be better to use fields, but that screws up jsdoc.

a typescript source wouldn't error here